src/bin/bench.rs

```diff
@@ -159,29 +165,151 @@ fn main() {
 
 fn run(config: Config) -> Result<(), Box<Error>> {
     println!("running benchmarking suite...");
+
+    // the thread pool we use to perform our operations on the tree
+    let pool = rayon::ThreadPool::new(rayon::Configuration::new().num_threads(config.num_threads))
+        .unwrap();
+
+    // create a default rsdb config
+    let rsdb_config = rsdb::Config::default();
+    let tree = rsdb_config.tree();
+
+    perform_tree_operations(&tree, &config, &pool);
+
     Ok(())
 }
 
+fn perform_tree_operations(tree: &Tree, config: &Config, pool: &ThreadPool) {
+    let sum_ops = config.set + config.scan + config.get + config.delete + config.cas;
+    let ops = vec![
+        (Op::Set, config.set),
+        (Op::Scan, config.scan),
+        (Op::Get, config.get),
+        (Op::Delete, config.delete),
+        (Op::Cas, config.cas),
+    ];
+
+    for i in 0..config.num_operations {
+        let op_to_perform = get_operation_choice(&ops, sum_ops);
+
+        let op: Box<Fn() + Sync> = match op_to_perform {
+            Some(&Op::Set) => Box::new(|| perform_set_operation(&tree)),
+            Some(&Op::Scan) => Box::new(|| perform_scan_operation(&tree)),
+            Some(&Op::Get) => Box::new(|| perform_get_operation(&tree)),
+            Some(&Op::Delete) => Box::new(|| perform_delete_operation(&tree)),
+            Some(&Op::Cas) => Box::new(|| perform_cas_operation(&tree)),
+            _ => Box::new(|| {}),
+        };
+
+        pool.install(|| op());
```

Comment:
this particular loop is probably faster than any individual operation in the tree, but I'm not sure if each iteration is like 32x faster, which is a common number of cores that are available to a server these days.